### PR TITLE
Re-introduce the single arg contract call optimisation for IR.

### LINKED ISSUE #1138: Re-enable all the checks in the `lib-std` tests
https://github.com/FuelLabs/sway/pull/1130#pullrequestreview-930346932 disables some parts of the `lib-std` tests because the SDK still requires a small change to become compatible with IR: https://github.com/FuelLabs/fuels-rs/issues/201

Uncomment all the disabled parts of the tests once we have a new SDK version with the required changes.

### PR DESCRIPTION
**Pull request by otrho**
Re-introduce the special case where if there's a single argument to a function call it is embedded directly in the call frame rather than being a pointer to the args elsewhere.

This will fix for errors we currently get when trying to call some contract from the SDK.

I'm halfway through working out what else needs to be fixed for contract calls + SDK but I've been on the road a bit and haven't had time.  So I thought I'd submit this PR now in case it's blocking anyone.

Closes #1138.

### CODE REVIEW

**Comment by otrho** [comment]
Argh, the E2E test is failing, I assume because the code expects the single arg optimisation but the IR doesn't compile them.  But I may be wrong... it's getting late for me, I'll revisit tomorrow.

**Review by nfurfaro — COMMENTED** [review]
Nice to see stdlib tests re-enabled! Needs another set of eyes on IR stuff as I'm not familiar enough with it to give a thorough review.

**Review by mohammadfawaz — APPROVED** [review]
I looked at this careful and I think it's reasonable. 
image

File: sway-core/src/optimize.rs [hunk 1]
```diff
@@ -822,46 +822,103 @@ impl FnCompiler {
...
+                // We're still undecided as to whether this should be decided by type or size.
+                // Going with type for now.
+                let arg0_type = arg0.get_type(context).unwrap();
+                if arg0_type.is_copy_type() {
+                    arg0
+                } else {
+                    // Copy this value to a new location.  This is quite inefficient but we need to
+                    // pass by reference rather than by value.  Optimisation passes can remove all
+                    // the unnecessary copying eventually, though it feels like we're jumping
+                    // through a bunch of hoops here (employing the single arg optimisation) for
```
Comment (sezna) [inline sway-core/src/optimize.rs:840]: @otrho is letting his political opinions leak into the code again

Comment (otrho) [reply sway-core/src/optimize.rs:840]: It's just an observation. No judgement. 🙂

**Pull request merged**